# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1298961341 28800
#      Mon Feb 28 22:35:41 2011 -0800
# Node ID bea828c03969d40c3f87a673774e429a2288e019
# Parent  bac54daffde2907739e5bd005e1f98e3677c5c1f
Add new isrow, iscolumn functions.

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2011-02-28  Rik  <octave@nomad.inbox5.com>
+
+	* NEWS: Mention new functions isrow, iscolumn.
+
 2011-02-23  Jordi Gutiérrez Hermoso  <jordigh@gmail.com>
 
 	* .hgignore: Cleanup and more files to be ignore that get added
 	to the source during build time.
 
 2011-02-22  Rik  <octave@nomad.inbox5.com>
 
 	* Makefile.am: New explicit 'make check' target.
diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -1,11 +1,17 @@
 Summary of important user-visible changes for version 3.6:
 ---------------------------------------------------------
 
+ ** The PCRE library is now required to build Octave.
+
+ ** New functions added.
+    iscolumn
+    issrow
+ 
  ** Deprecated functions.
 
     The following functions were deprecated in Octave 3.2 and have been
     removed from Octave 3.6.
                                            
       create_set          spcholinv    splu   
       dmult               spcumprod    spmax
       iscommand           spcumsum     spmin
@@ -13,18 +19,16 @@ Summary of important user-visible change
       lchol               spdiag       spqr
       loadimage           spfind       spsum
       mark_as_command     sphcat       spsumsq
       mark_as_rawcommand  spinv        spvcat 
       spatan2             spkron       str2mat
       spchol              splchol      unmark_command
       spchol2inv          split        unmark_rawcommand
 
-  ** The PCRE library is now required to build Octave.
-
 Summary of important user-visible changes for version 3.4:
 ---------------------------------------------------------
 
  ** BLAS and LAPACK libraries are now required to build Octave.  The
     subset of the reference BLAS and LAPACK libraries has been removed
     from the Octave sources.
   
  ** The ARPACK library is now distributed with Octave so it no longer
diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,12 @@
+2010-02-28  Rik  <octave@nomad.inbox5.com>
+
+	* interpreter/numbers.txi: Add isrow, iscolumn functions to manual.
+
 2011-02-25  Jordi Gutiérrez Hermoso  <jordigh@gmail.com>
 
 	* faq/OctaveFAQ.texi: Uniformise all mentions of Matlab to be
 	typeset in the @sc{Matlab} style. Also do some minor
 	proofreading like removing trailing spaces and some minor
 	spellchecking.
 
 2010-02-19  Rik  <octave@nomad.inbox5.com>
diff --git a/doc/interpreter/numbers.txi b/doc/interpreter/numbers.txi
--- a/doc/interpreter/numbers.txi
+++ b/doc/interpreter/numbers.txi
@@ -814,16 +814,20 @@ variable.
 @DOCSTRING(isfloat)
 
 @DOCSTRING(iscomplex)
 
 @DOCSTRING(ismatrix)
 
 @DOCSTRING(isvector)
 
+@DOCSTRING(isrow)
+
+@DOCSTRING(iscolumn)
+
 @DOCSTRING(isscalar)
 
 @DOCSTRING(issquare)
 
 @DOCSTRING(issymmetric)
 
 @DOCSTRING(ishermitian)
 
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-02-28  Rik  <octave@nomad.inbox5.com>
+
+	* general/iscolumn.m, general/isrow.m: Add 2 new utility functions
+	to check for row or column vector.
+
 2010-02-28  Rik  <octave@nomad.inbox5.com>
 
 	* sparse/treeplot.m: Use 'o' plot style as default for nodes
 
 2010-02-27  Rik  <octave@nomad.inbox5.com>
 
 	* special-matrix/pascal.m: Fix incorrect statement in documentation
 	for pascal.m.  Bug #32523.
diff --git a/scripts/general/iscolumn.m b/scripts/general/iscolumn.m
new file mode 100644
--- /dev/null
+++ b/scripts/general/iscolumn.m
@@ -0,0 +1,62 @@
+## Copyright (C) 2011 John W. Eaton
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {} iscolumn (@var{x})
+## Return true if @var{x} is a column vector.
+## @seealso{isrow, isscalar, isvector, ismatrix}
+## @end deftypefn
+
+## Author: Rik Wehbring
+
+function retval = iscolumn (x)
+
+  retval = false;
+
+  if (nargin != 1)
+    print_usage ();
+  endif
+
+  sz = size (x);
+  retval = (ndims (x) == 2 && (sz(2) == 1));
+
+endfunction
+
+%!assert(iscolumn ([1, 2, 3]), false);
+%!assert(iscolumn ([1; 2; 3]));
+%!assert(iscolumn (1));
+%!assert(iscolumn ([]), false);
+%!assert(iscolumn ([1, 2; 3, 4]), false);
+
+%!test
+%! warning ("off", "Octave:str-to-num");
+%! assert((iscolumn ("t")));
+%!test
+%! warning ("off", "Octave:str-to-num");
+%! assert(!(iscolumn ("test")));
+
+%!assert(!(iscolumn (["test"; "ing"])));
+
+%!test
+%! s.a = 1;
+%! assert((iscolumn (s)));
+
+%% Test input validation
+%!error iscolumn ();
+%!error iscolumn ([1, 2], 2);
+
diff --git a/scripts/general/isrow.m b/scripts/general/isrow.m
new file mode 100644
--- /dev/null
+++ b/scripts/general/isrow.m
@@ -0,0 +1,62 @@
+## Copyright (C) 2011 John W. Eaton
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {} isrow (@var{x})
+## Return true if @var{x} is a row vector.
+## @seealso{iscolumn, isscalar, isvector, ismatrix}
+## @end deftypefn
+
+## Author: Rik Wehbring
+
+function retval = isrow (x)
+
+  retval = false;
+
+  if (nargin != 1)
+    print_usage ();
+  endif
+
+  sz = size (x);
+  retval = (ndims (x) == 2 && (sz(1) == 1));
+
+endfunction
+
+%!assert(isrow ([1, 2, 3]));
+%!assert(isrow ([1; 2; 3]), false);
+%!assert(isrow (1));
+%!assert(isrow ([]), false);
+%!assert(isrow ([1, 2; 3, 4]), false);
+
+%!test
+%! warning ("off", "Octave:str-to-num");
+%! assert((isrow ("t")));
+%!test
+%! warning ("off", "Octave:str-to-num");
+%! assert((isrow ("test")));
+
+%!assert(!(isrow (["test"; "ing"])));
+
+%!test
+%! s.a = 1;
+%! assert((isrow (s)));
+
+%% Test input validation
+%!error isrow ();
+%!error isrow ([1, 2], 2);
+
diff --git a/scripts/general/module.mk b/scripts/general/module.mk
--- a/scripts/general/module.mk
+++ b/scripts/general/module.mk
@@ -39,19 +39,21 @@ general_FCN_FILES = \
   general/interp1.m \
   general/interp1q.m \
   general/interp2.m \
   general/interp3.m \
   general/interpn.m \
   general/interpft.m \
   general/is_duplicate_entry.m \
   general/isa.m \
+  general/iscolumn.m \
   general/isdir.m \
   general/isequal.m \
   general/isequalwithequalnans.m \
+  general/isrow.m \
   general/isscalar.m \
   general/issquare.m \
   general/isvector.m \
   general/loadobj.m \
   general/logspace.m \
   general/nargchk.m \
   general/nargoutchk.m \
   general/nextpow2.m \
